var/spack/repos/builtin/packages/helics/package.py

```diff
@@ -0,0 +1,124 @@
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+# ----------------------------------------------------------------------------
+# If you submit this package back to Spack as a pull request,
+# please first remove this boilerplate and all FIXME comments.
+#
+# This is a template package file for Spack.  We've put "FIXME"
+# next to all the things you'll want to change. Once you've handled
+# them, you can save this file and test your package like this:
+#
+#     spack install helics
+#
+# You can edit this file again by typing:
+#
+#     spack edit helics
+#
+# See the Spack documentation for more information on packaging.
+# ----------------------------------------------------------------------------
+
+from spack import *
+
+
+class Helics(CMakePackage):
+    """HELICS is a general-purpose, modular, highly-scalable co-simulation
+    framework that runs cross-platform (Linux, Windows, and Mac OS X) and
+    supports both event driven and time series simulation."""
+
+    homepage = "https://github.com/GMLC-TDC/HELICS"
+    url      = "https://github.com/GMLC-TDC/HELICS/releases/download/v2.4.1/Helics-v2.4.1-source.tar.gz"
+    git      = "https://github.com/GMLC-TDC/HELICS.git"
+
+    maintainers = ['nightlark']
+
+    version('mpi-fix', branch='fix-openmpi-cxx', submodules=True)
+    version('develop', branch='develop', submodules=True)
+    version('master', branch='master', submodules=True)
+    version('2.4.2', sha256='957856f06ed6d622f05dfe53df7768bba8fe2336d841252f5fac8345070fa5cb')
+    version('2.4.1', sha256='ac077e9efe466881ea366721cb31fb37ea0e72a881a717323ba4f3cdda338be4')
+
+    variant('build_type', default='Release',
+            description='CMake build type',
+            values=('Debug', 'Release', 'RelWithDebInfo', 'MinSizeRel'))
+    variant('apps', default=True, description="Install the HELICS apps")
+    variant('c_shared', default=True, description="Install the C shared library")
+    variant('cxx_shared', default=True, description="Install the CXX shared library")
+    variant('zmq', default=True, description="Enable ZeroMQ core types")
+    variant('tcp', default=True, description="Enable TCP core types")
+    variant('udp', default=True, description="Enable UDP core type")
+    variant('ipc', default=True, description="Enable IPC core type")
+    variant('inproc', default=True, description="Enable in-process core type")
+    variant('mpi', default=False, description="Enable MPI core type")
+    variant('boost', default=True, description="Compile with Boost libraries")
+    variant('asio', default=True, description="Compile with ASIO libraries")
+    variant('swig', default=False, description="Build language bindings with SWIG")
+    variant('webserver', default=True, description="Enable the integrated webserver in the HELICS broker server")
+
+    # Build dependency
+    for _version in ['@master', '@develop']:
+        depends_on('git', type='build', when=_version)
```

Comment:
You can instead do:
```python
depends_on('git', type='build', when='@master:')
```
The open version range means "master or higher". `@master,develop` is also valid.

Reply:
Okay, so other branches (e.g. "test-branch") would be considered higher than master (and develop)?

Reply:
Nope, see #1983. There is special handling for develop, master, head, and trunk. Any other non-numeric version is sorted as being older than all other numeric versions.

Reply:
Okay, got it.